File: VCL/C++Builder/TigerGeocoding/InfoForm.cpp
//=============================================================================
// This source code is a part of TatukGIS Developer Kernel.
// (c)2000-2025 TatukGIS. ALL RIGHTS RESERVED.
//=============================================================================

//  How to provide Geocoding using name standarization
//
//  Check project\options\directories in a case of any problems during compilation
//---------------------------------------------------------------------------


#include <vcl.h>
#pragma hdrstop

#include "InfoForm.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma resource "*.dfm"
TFormInfo *FormInfo;
//---------------------------------------------------------------------------
__fastcall TFormInfo::TFormInfo(TComponent* Owner)
        : TForm(Owner)
{
}
//---------------------------------------------------------------------------

void __fastcall TFormInfo::FormClose(TObject *Sender, TCloseAction &Action)
{
  Form1->sgrdMemo->Hint = "Double click for info";
}
//---------------------------------------------------------------------------

void __fastcall TFormInfo::FormShow(TObject *Sender)
{
  Form1->sgrdMemo->Hint = "";
}
//---------------------------------------------------------------------------

File: VCL/Delphi/TigerGeocoding/InfoForm.pas
//=============================================================================
// This source code is a part of TatukGIS Developer Kernel.
//=============================================================================
{
  How to provide Geocoding using name standardization
}
unit InfoForm;

interface

uses
  System.SysUtils,
  System.Classes,

  Winapi.Windows,
  Winapi.Messages,

  Vcl.Graphics,
  Vcl.Controls,
  Vcl.Forms,
  Vcl.Dialogs,
  Vcl.Grids;

type
  TFormInfo = class(TForm)
    sgrdInfo: TStringGrid;
    procedure FormShow(Sender: TObject);
    procedure FormClose(Sender: TObject; var Action: TCloseAction);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  FormInfo: TFormInfo;

implementation

uses MainForm;

{$R *.DFM}

procedure TFormInfo.FormShow(Sender: TObject);
begin
  Form1.sgrdMemo.Hint := '';
end;

procedure TFormInfo.FormClose(Sender: TObject; var Action: TCloseAction);
begin
  Form1.sgrdMemo.Hint := 'Double click for info';
end;

//==================================== END =====================================
end.
